generator/src/generate/disasm.rs

```diff
@@ -17,7 +17,10 @@ use crate::{
 
 pub fn generate_disasm(isa: &Isa, isa_args: &IsaArgs, max_args: usize) -> Result<TokenStream> {
     // Generate opcode enum and mnemonics array
-    let (opcode_enum_tokens, opcode_mnemonics_tokens, num_opcodes_token) = generate_opcode_tokens(&isa.opcodes);
+    let (opcode_enum_tokens, opcode_mnemonics_tokens, num_opcodes) = generate_opcode_tokens(&isa.opcodes);
+    assert!(num_opcodes <= 256);
```

Comment:
Just noticed this bug, present in my ppc750cl commit as well
```suggestion
    assert!(num_opcodes < 256);
```

Reply:
`opcode_max` is `num_opcodes - 1`, so it should be fine since 255 fits in a `u8`

Reply:
We reserve 255 for `Opcode::Illegal`

Reply:
Oh huh, I didn't realize the ppc implementation did that.
Apparently the arm and thumb ones are implemented differently, and reserve 0 for `Opcode::Illegal`.
Even though the list of opcodes starts at the first non-illegal one (adc), I think this line:
```rs
let num_opcodes = sorted_opcodes.len() + 1;
```
Handles accounting for that one extra one.
So it should be okay for unarm?

Reply:
Ah, I think your interpretation is correct then!